launcher/modplatform/modpacksch/FTBPackInstallTask.cpp

```diff
@@ -302,12 +302,9 @@ void PackInstallTask::install()
     auto components = instance.getPackProfile();
     components->buildingFromScratch();
 
-    for (auto target : m_version.targets) {
-        if (target.type == "game" && target.name == "minecraft") {
-            components->setComponentVersion("net.minecraft", target.version, true);
-            break;
-        }
-    }
+    auto target_it = std::find_if(m_version.targets.cbegin(), m_version.targets.cend(), [](const auto& target) { return target.type == "game" && target.name == "minecraft"; });
+    if (target_it != m_version.targets.cend())
```

Comment:
You could put `target_it` in the if statement's initializer.

Reply:
No real benefit from doing this as we don't target C++17, so we can't use if statements with initializers (I swear we did, but CMake won't let me build it so I'm assuming we don't)

Reply:
We do target C++17

Reply:
Nevermind, clangd was just being stupid. Many such cases!